ios/OTSessionManager.swift

```diff
@@ -28,7 +28,7 @@ class OTSessionManager: RCTEventEmitter {
   
   @objc override func supportedEvents() -> [String] {
     let allEvents: [String] = ["\(sessionPreface)streamCreated", "\(sessionPreface)streamDestroyed", "\(sessionPreface)sessionDidConnect", "\(sessionPreface)sessionDidDisconnect", "\(sessionPreface)connectionCreated", "\(sessionPreface)connectionDestroyed", "\(sessionPreface)didFailWithError", "\(publisherPreface)streamCreated", "\(sessionPreface)signal", "\(publisherPreface)streamDestroyed", "\(publisherPreface)didFailWithError", "\(publisherPreface)audioLevelUpdated", "\(subscriberPreface)subscriberDidConnect", "\(subscriberPreface)subscriberDidDisconnect", "\(subscriberPreface)didFailWithError", "\(subscriberPreface)videoNetworkStatsUpdated", "\(subscriberPreface)audioNetworkStatsUpdated", "\(subscriberPreface)audioLevelUpdated", "\(subscriberPreface)subscriberVideoEnabled", "\(subscriberPreface)subscriberVideoDisabled", "\(subscriberPreface)subscriberVideoDisableWarning", "\(subscriberPreface)subscriberVideoDisableWarningLifted", "\(subscriberPreface)subscriberVideoDataReceived", "\(sessionPreface)archiveStartedWithId", "\(sessionPreface)archiveStoppedWithId", "\(sessionPreface)sessionDidBeginReconnecting", "\(sessionPreface)sessionDidReconnect"];
-    return allEvents
+    return allEvents + jsEvents
```

Comment:
Can you explain why we need to append `jsEvents`? and what would that possibly be? As here, it's only an empty array.

Reply:
The events are added to the `jsEvents` array using the `setNativeEvents` method that's invoked from the JS layer.
https://github.com/opentok/opentok-react-native/blob/master/src/OT.js#L33
https://github.com/opentok/opentok-react-native/blob/master/ios/OTSessionManager.swift#L143

Here's some context:
For iOS, React Native verifies that an event is supported before it's dispatched via the`RCTEventEmitter` class. This is why we have a list of events that we support in the `allEvents` array, but since we support multiple publishers, we need to track the events separately for each publisher so we create a `uuid` for each publisher and add that as a preface to the event names when setting these events using `setNativeEvents`. This is why this error was coming up: https://user-images.githubusercontent.com/1220480/40053681-d453741c-585f-11e8-94be-5cb15944aea3.png 

Hope this helps!